src/ResourceManager/RecoveryServices/Commands.RecoveryServices.SiteRecovery.Test/ScenarioTests/A2A/AsrA2ATestsBase.cs

```diff
@@ -0,0 +1,235 @@
+// ----------------------------------------------------------------------------------
+// 
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net.Security;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.Azure.Commands.Common.Authentication;
+using Microsoft.Azure.Management.Internal.Resources;
+using Microsoft.Azure.Management.RecoveryServices;
+using Microsoft.Azure.Management.RecoveryServices.SiteRecovery;
+using Microsoft.Azure.Test;
+using Microsoft.Azure.Test.HttpRecorder;
+using Microsoft.WindowsAzure.Commands.ScenarioTest;
+using Microsoft.WindowsAzure.Commands.Test.Utilities.Common;
+using RestTestFramework = Microsoft.Rest.ClientRuntime.Azure.TestFramework;
+using Microsoft.Azure.Internal.Subscriptions;
+using Microsoft.Azure.ServiceManagemenet.Common.Models;
+using Microsoft.Azure.Commands.Common.Compute.Version_2018_04;
+
+namespace RecoveryServices.SiteRecovery.Test
+{
+    public abstract class AsrA2ATestsBase : RMTestBase
+    {
+        protected string vaultSettingsFilePath;
+        protected string powershellFile;
+        protected string powershellHelperFile;
+        private CSMTestEnvironmentFactory csmTestFactory;
+        private EnvironmentSetupHelper helper;
+
+        protected AsrA2ATestsBase()
+        {
+        }
+
+        protected void initialize()
+        {
+            this.helper = new EnvironmentSetupHelper();
+        }
+
+        public ResourceManagementClient RmRestClient { get; private set; }
+        
+        
+        public RecoveryServicesClient RecoveryServicesMgmtClient { get; private set; }
+        public SiteRecoveryManagementClient SiteRecoveryMgmtClient { get; private set; }
+
+        public SubscriptionClient SubscriptionClient { get; private set; }
```

Comment:
Same comment at the other test controller above. I don't believe this should be needed.